### PR TITLE
Fix double free in main.cpp's save

### LINKED ISSUE #138: Not working scale multiple of x4 or more.
Like #137, there is a multiple of x4 or more in the scale command, but the image is not output.

### LINKED ISSUE #140: No output file
- program version: 20210210(https://github.com/nihui/waifu2x-ncnn-vulkan/commit/77534d94b562148d07767c5c647a6a8b26013e2e)
- windows 20H2
- intel UHD Graphics 620  driver version: 27.20.100.9168
- nVidia GTX 1050 2GB   driver version: 27.21.14.6172 in task manager and 461.72 in nVidia Geforce experience

#### After seeing other issues, I've checked that...
- ...if I use GPU 0, the dedicated GPU RAM was occupied only roughly 0.5 GB
- ...if I use GPU 1, shared GPU RAM was occupied 1.4 GB while the total is 3.9 GB
- ...it didn't run and close immediately. The time was different between hardware.
- ...I'm using full path name as showing below.

### Command Lines:
- waifu2x-ncnn-vulkan.exe -i "D:\Documents\temp\video2x-test-pic\extracted_2146.jpg" -o "D:\Documents\temp\video2x-test-pic-out\2146.jpg" -n 3 -s 4 **-g -1 -j 1:4:1** -f jpg -v
- waifu2x-ncnn-vulkan.exe -i "D:\Documents\temp\video2x-test-pic\extracted_2146.jpg" -o "D:\Documents\temp\video2x-test-pic-out\2146.jpg" -n 3 -s 4 **-g 0 -j 1:2:1** -f jpg -v
- waifu2x-ncnn-vulkan.exe -i "D:\Documents\temp\video2x-test-pic\extracted_2146.jpg" -o "D:\Documents\temp\video2x-test-pic-out\2146.jpg" -n 3 -s 4 **-g 1 -j 1:1:1** -f jpg -v
### Screenshot
未命名 - 3

### PR DESCRIPTION
**Pull request by utybo**
I should preface this by saying that **I do not know if this is the right solution**. This is probably hacky, as I do not know why a free was done there in the first place.

This is a follow-up to [this comment I made](https://github.com/nihui/waifu2x-ncnn-vulkan/issues/138#issuecomment-839254692) on #138.

This *should* fix #138 and fix #140.

### CODE REVIEW

**Comment by utybo** [comment]
@nihui Could you please give the workflow an approval? I'd like to know if this breaks the CI or not.

**Comment by ArchieMeng** [comment]
@utybo I am not sure, but I feel like you might just omit the release of ncnn::Mat or the image memory space. And I am afraid this may leads to memory leak issues.
Additionally, I remember ncnn had fixed the problem in the recently commits. Could you try update ncnn to the latest version and build again to see if the problem still exists? Because I don't have the problem you have and I have no way to see if this works.

**Comment by utybo** [comment]
@ArchieMeng I can still repro using the latest version of ncnn (tag `20210507`). The output is the same as [my initial comment](https://github.com/nihui/waifu2x-ncnn-vulkan/issues/138#issuecomment-839254692)

> I am not sure, but I feel like you might just omit the release of ncnn::Mat or the image memory space. And I am afraid this may leads to memory leak issues.

To be honest, I don't really know what the loop does in the first place, but from what I understand, the freeing is unecessary as it is already done by getting out of the scope where the `Task` variable is declared (which is a value and not a pointer). The (implicit) destructor of `Task` automatically calls [`ncnn::Mat`'s destructor](https://github.com/Tencent/ncnn/blob/ae14ca91ac9eedf590ca12d3dff4810d98027b65/src/mat.h#L751), which in turns calls [`ncnn::Mat::release`](https://github.com/Tencent/ncnn/blob/ae14ca91ac9eedf590ca12d3dff4810d98027b65/src/mat.h#L995) which then does [the problematic free for us](https://github.com/Tencent/ncnn/blob/ae14ca91ac9eedf590ca12d3dff4810d98027b65/src/mat.h#L999-L1002). `stbi_image_free` is a macro which ends up leading to a `free`, so all branches of both if/elses (the one in code and the one using the preprocessor) end up doing the same thing, which is `free`ing `v.inimage.data`.

TL;DR: the code frees an internal pointer of `ncnn::Mat` which is already destroyed by `Task`'s destructor, which calls `ncnn::Mat`'s destructor, which takes care of that internal pointer.

**Comment by nihui** [comment]
fixed in https://github.com/nihui/waifu2x-ncnn-vulkan/commit/f332f3c256d3a7f9b7c18a23a05a4e402ea9fd18
